synapse/handlers/profile.py

```diff
@@ -162,18 +166,32 @@ def _update_join_states(self, requester):
 
         for room_id in room_ids:
             handler = self.hs.get_handlers().room_member_handler
+            member_event = yield handler.get_member_event(user, room_id)
```

Comment:
I believe you want `StateHandler.get_current_state` for this. See https://github.com/matrix-org/synapse/blob/develop/synapse/api/auth.py#L123 for example.